gprofiler/profilers/java.py

```diff
@@ -259,11 +284,107 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": self.get_jvm_flags(process)}
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    @functools.lru_cache(maxsize=1024)
+    def get_jvm_flags(self, process: Process) -> Optional[List[tuple]]:
+        jvm_raw_flags = self.get_jvm_flags_raw(process)
+        if jvm_raw_flags is None:
+            return None
+
+        return [dataclasses.astuple(flag) for flag in jvm_raw_flags if self.filter_jvm_flags(flag)]
+
+    @staticmethod
+    def filter_jvm_flags(flag: JvmFlag) -> bool:
+        if flag.flag_origin == "default":
+            return False
+
+        if flag.flag_type not in [
+            "bool",
+            "int",
+            "uint",
+            "intx",
+            "uintx",
+            "uint64_t",
+            "size_t",
+            "double",
+        ]:
+            return False
+
+        if set(flag.flag_kind).intersection({"notproduct", "pd", "experimental", "develop"}):
+            return False
+
+        return True
+
+    def get_jvm_flags_raw(self, process: Process) -> Optional[List[JvmFlag]]:
+        try:
+            output = self.jattach_jcmd_runner.run(process, "VM.flags -all")
+        except CalledProcessError as e:
+            logging.warning(f"Couldn't get jvm flags for process {process.pid}: {e.stderr}")
+            return None
+
+        vm_flags = []
+        # The output of VM.flags -all format on jdk 8:
+        # bool UseCompressedClassPointers               := true                                {lp64_product}
+        # flag_type flag_name                           := flag_value                          {flag_kind}
+        # ":=" indicates non default origin for the flag, while "=" indicates default origin
+
+        # The output of VM.flags -all format on jdk 9+:
+        # bool OptoScheduling                           = false                               {C2 pd product} {default}
+        # flag_type flag_name                           = flag_value                          {flag_kind} {flag_origin}
+
+        # flag_kind is space separated list of kinds, e.g. "C2 pd product"
+
+        # possible flag kinds: "product", "manageable", "diagnostic", "experimental", "notproduct", "develop",
+        # "lp64_product", "rw", "pd", "JVMCI", "C1", "C2", "ARCH"
+
+        # possible flag types: "bool", "int", "uint", "intx", "uintx", "uint64_t", "size_t", "double", "ccstr",
+        # "ccstrlist"
+
+        # possible flag origins: default, non-default, command line, environment, config file, management, ergonomic,
+        # attach, internal, jimage
+
+        vm_flags_pattern = re.compile(
+            r"(?P<flag_type>\S+)\s+(?P<flag_name>\S+)\s+(?P<flag_is_non_default_origin_only_jdk_8>:)?= (?P<flag_value>\S*)\s+{(?P<flag_kind>.+?)}(?:\s*{(?P<flag_origin_jdk_9>.*)})*"  # noqa: E501
```

Comment:
```suggestion
            r"(?P<flag_type>\S+)\s+(?P<flag_name>\S+)\s+(?P<flag_is_non_default_origin_only_jdk_8>:)?= (?P<flag_value>\S*)\s+{(?P<flag_kind>.+?)}(?:\s*{(?P<flag_origin_jdk_9>.*)})?"  # noqa: E501
```
Doesn't the origin appear at most once?